src/main/java/com/desiato/puresynth/validators/AbstractValidator.java

```diff
@@ -0,0 +1,28 @@
+package com.desiato.puresynth.validators;
+
+import com.desiato.puresynth.exceptions.ErrorMessage;
+import com.desiato.puresynth.exceptions.ValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class AbstractValidator<T> {
+
+    private final List<ErrorMessage> errorMessages = new ArrayList<>();
```

Comment:
Do you need to maintain this state? Maybe it would be enough to create a new list in the beginning of `validate` method? Then you wouldn't need to reset these errors. 

Also, this code probably won't work in a multithreaded environemnt. If you weere to handle multiple requests at once, you could start populating/clearing there errors at the same time. You have race conditions here -> it wouldn't work.